django/contrib/admin/options.py

```diff
@@ -2271,7 +2271,7 @@ def history_view(self, request, object_id, extra_context=None):
         app_label = self.opts.app_label
         action_list = (
             LogEntry.objects.filter(
-                object_id=unquote(object_id),
+                object_id=self.unquote(object_id),
```

Comment:
`LogEntry` struggles to get the admin url for models with composite primary keys

![Screenshot from 2024-08-21 12-07-34](https://github.com/user-attachments/assets/43a10928-c4c1-4463-a608-5a2eb45d87b9)

This is because `object_id` saves as a string field rather than a tuple
It might be worth `LogEntry.get_admin_url` to return `None` for models with composite primary keys unless you can think of a nice way to handle it?

Reply:
This is the same problem as [generic foreign keys](https://github.com/csirmazbendeguz/django/pull/9). As a side note, I wonder why `LogEntry` doesn't use a `GenericForeignKey`.

My proposal is to store the composite primary keys in a `CharField` / `TextField` as a JSON array.

I initially thought this class used a `GenericForeignKey`, I was planning to address that in a separate PR, especially because I wanted a separate review and approval on the JSON approach.

Would you like me to adjust `LogEntry` in this PR?

Reply:
Perhaps in this PR it returns None (which I think means there is no link) and then in the `GenericForeignKey` one, it becomes a valid link?

Reply:
My concern is `LogEntry`s made between merging this PR and the follow-up PR won't work. So maybe it would be better to skip creating log entries for models with composite pks for now?

Reply:
Ah, understood :thinking: .....errrrrr, we could remove admin support from this PR and add an error similar to the abstract models error:
```python
if model._meta.abstract:
    raise ImproperlyConfigured(
        "The model %s is abstract, so it cannot be registered with admin." % model.__name__
    )
```
Document that you cannot register models with a composite primary key in the admin and then this work is in a separate PR?

I have no idea how important `LogEntry` is and whether having holes in the data is "ok" :thinking:

Reply:
Yeah, that makes sense, I'll adjust. Thanks @sarahboyce !

Reply:
I have removed admin support from this PR, and added it in a [new PR](https://github.com/csirmazbendeguz/django/pull/10).

Reply:
I will adjust the `LogEntry` once we've agreed on the generic foreign key approach.
So I believe admin support will have to wait until generic foreign key support has been added.

Reply:
Thank you, I think this makes sense to do :star: